src/tests/ftest/daos_test/daos_core_test-faults.yaml

```diff
@@ -0,0 +1,106 @@
+# change host names to your reserved nodes, the
+# required quantity is indicated by the placeholders
+hosts:
+  test_servers:
+    - server-A
+    - server-B
+    - server-C
+    - server-D
+# Note that subtests below can set their own timeout so this
+# should be a general average of all tests not including outliers
+# (I'm looking at you "rebuild tests")
+timeout: 600
+pool:
+  #This will create 8G of SCM and 16G of NVMe size of pool.
+  scm_size: 8G
+server_config:
+  servers_per_host: 2
+  name: daos_server
+  servers:
+    0:
+      pinned_numa_node: 0
+      nr_xs_helpers: 1
+      fabric_iface: ib0
+      fabric_iface_port: 31416
+      log_file: daos_server0.log
+      bdev_class: nvme
+      bdev_list: ["0000:81:00.0"]
+      scm_class: dcpm
+      scm_list: ["/dev/pmem0"]
+      scm_mount: /mnt/daos0
+    1:
+      pinned_numa_node: 1
+      nr_xs_helpers: 1
+      fabric_iface: ib1
+      fabric_iface_port: 31417
+      log_file: daos_server1.log
+      bdev_class: nvme
+      bdev_list: ["0000:da:00.0"]
+      scm_class: dcpm
+      scm_list: ["/dev/pmem1"]
+      scm_mount: /mnt/daos1
+daos_tests:
+  num_clients:
+    num_clients: 1
+  num_replicas:
+    num_replicas: 1
+  Tests: !mux
+    test_d:
+      daos_test: d
+      test_name: DAOS degraded-mode tests
+    test_m:
+      daos_test: m
+      test_name: Management tests
+    test_p:
+      daos_test: p
+      test_name: Pool tests
+    test_c:
+      daos_test: c
+      test_name: DAOS container tests
+    test_e:
+      daos_test: e
+      test_name: DAOS epoch tests
+    test_X:
+      daos_test: X
+      test_name: DAOS DTX tests
+    test_V:
+      daos_test: V
+      test_name: DAOS verify consistency tests
+      test_timeout: 900
+    test_i:
+      daos_test: i
+      test_name: IO test
+      test_timeout: 900
+    test_A:
+      daos_test: A
+      test_name: DAOS Object Array tests
+    test_D:
+      daos_test: D
+      test_name: DAOS Array tests
+    test_K:
+      daos_test: K
+      test_name: DAOS KV tests
+    test_F:
+      daos_test: F
+      test_name: DAOS File System tests
+    test_C:
+      daos_test: C
+      test_name: DAOS capability tests
+    test_o:
+      daos_test: o
+      test_name: Epoch recovery tests
+    test_R:
+      daos_test: R
+      test_name: DAOS MD replication tests
+    test_v:
+      daos_test: v
+      test_name: DAOS rebuild simple tests
+      test_timeout: 1500
+    test_O:
+      daos_test: O
+      test_name: OID Allocator tests
+      test_timeout: 1350
+    test_z:
+      daos_test: z
+      test_name: Checksum tests
+      test_timeout: 900
```

Comment:
All of these tests above require fault-injection?  I didn't know we had introduced FI into so many daos_test tests.

But if they do all require FI, shouldn't they be removed from [daos_core_test.yaml](https://github.com/daos-stack/daos/blob/74a7c5c1061b773b8095ad9d298ae733d66038a8/src/tests/ftest/daos_test/daos_core_test.yaml)?

Reply:
I removed the subtests from each daos_test that expect an error to occur using the -u subtests param of daos_test.    For instance in the daos_test -m, only subtest MGMT5 injects an error, so MGMT1-4 should still pass with faults disable.   In the checksum subtests all failed except for 2 with faults disabled.

Yes there are quite a few that use daos_fail_loc_set which would define a fault to inject.   The error will not get injected unless daos_fail_check() has verified that the specific error has been setup with daos_fail_loc_set.   David Q. has a PR which will prevent the error from being injected by daos_fail_check() if !d_fault_inject_enabled().   I used that PR branch to verify which daos_test subtests failed with !d_fault_inject_enabled()

Reply:
So, for example:
https://github.com/daos-stack/daos/blob/74a7c5c1061b773b8095ad9d298ae733d66038a8/src/tests/ftest/daos_test/daos_core_test.yaml#L51-L55
only runs the non-FI tests.  Agreed.  But you are also asking for all four of those tests to be run in the FI daos_test run:
https://github.com/daos-stack/daos/blob/74a7c5c1061b773b8095ad9d298ae733d66038a8/src/tests/ftest/daos_test/daos_core_test-faults.yaml#L51-L53

Given that every test run, except for RC test runs, will run both `daos_core_test-faults.yaml` and `daos_core_test.yaml` there is going to be duplicate tests running.

Ultimately, it seems that any test that is in `daos_core_test-faults.yaml` should then be excluded in `daos_core_test.yaml`, yes?

Reply:
Could we use the tags in jenkinsfile and do something like this:
 RC the tag would be; 
test_tag=xxxx,-faults
and for the non-RC it would be
test_tag=xxxx,faults

Reply:
I think the problem is that the structure depends on how the `Jenkinsfile` changes are implemented.  It can either be two separate stages where the FI stages are disabled during RCs or the tags can just be changed for existing stages as you suggest.

But that does raise the point that this change is missing the `Jenkinsfile` changes needed to effect the disabling of FI tests when running on an RC.